paddle/fluid/framework/parallel_executor.cc

```diff
@@ -63,7 +64,17 @@ ParallelExecutor::ParallelExecutor(
   member_->global_scope_ = scope;
   member_->use_cuda_ = exec_strategy.use_cuda_;
 
-  // Step 1. Bcast the params to devs.
+  // Step 1. check all memory usages of all places.
+  // This step will create BuddyAllocator for each place, which will
+  //    1. enforce that the place is avaliable and NOT used.
+  //    2. avoid ncclBcast hanging up for NOT enough memory to use.
+  for (size_t i = 0; i < member_->places_.size(); ++i) {
```

Comment:
Does these lines was intended to let the process fail fast?

Reply:
yes